client/src/components/frontend/Utility/ShareBar.js

```diff
@@ -1,34 +1,78 @@
 import React, { Component, PropTypes } from 'react';
 import { Link } from 'react-router';
+import { TwitterButton, FacebookButton } from 'react-social';
 
 export default class ShareBar extends Component {
 
   static displayName = "Utility.ShareBar";
 
   static propTypes = {
-    url: PropTypes.string
+    url: PropTypes.string,
+    twitterAppId: PropTypes.string,
+    facebookAppId: PropTypes.string
   };
 
+  constructor() {
+    super();
+    this.state = {
+      inBrowser: false
+    };
+  }
+
+  componentDidMount() {
+    // This won't be run by the server, so set an instance variable here
+    // that will be hidden otherwise
+    this.setState({
+      inBrowser: true
+    });
+  }
+
   render() {
+    if (!this.state.inBrowser) return null;
+
+    const twitterWindowOptions = ["", "", "width=600,height=300"];
+    const twitterAppId = this.props.twitterAppId ? this.props.twitterAppId : null;
+    const facebookAppId = this.props.facebookAppId ? this.props.facebookAppId : null;
+    {/* Todo: The hostname needs to be inserted here */}
```

Comment:
Is this todo still necessary? It's probably fine to get it from window.location, right?